### ISSUE TITLE
Bug: OidcAuth plugin force resets 'organisation' every login

### ISSUE THREAD

**Issue by jamesacris** [MISP/MISP#9478]
### Actual behavior

https://github.com/MISP/MISP/blob/532e5aba7b52410c844e31fb05bd9cfe7a8a41a2/app/Plugin/OidcAuth/Controller/Component/Auth/OidcAuthenticate.php#L16

Whatever this field is set to becomes the organisation of every user who logs in with OIDC. Even when the organisation is changed by an admin in MISP, it's reverted to `default_org` the next time the user logs in with OIDC.

### Expected behavior

The `default_org` is the organisation for _new_ users, but then the organisation set in MISP takes priority.

### Steps to reproduce

Install OidcAuth plugin as in docs [here](https://github.com/MISP/MISP/tree/532e5aba7b52410c844e31fb05bd9cfe7a8a41a2/app/Plugin/OidcAuth#readme).
Set up an external auth provider e.g. Keycloak (what I am using) and add misp as a client.
Add users with the correct realm roles as shown in the docs, then log into misp using OIDC.

### Version

2.4.179

### Operating System

Debian

### Operating System version

11

### PHP version

7.4.33

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jamesacris** [comment]
Hi,

wondering if there's been any progress on this?

**Comment by efx-sampath-ganji** [comment]
We are seeing similar issue.

**Comment by bita-kea** [comment]
same here. 
Our MISP has a lot of organisations. And each login (via keycloak) of existing users changes their organisation to the one specified in ‘default_org’. 
Is there any way to specify that the existing organisation in MISP would be prioritised? 
Or specify the organisation as we do with roles? 
So, then add the organisation field in the keycloak profile?

For context:
Keycloak v26

MISP v2.5
Ubuntu 24
php v8.3 
apache2 v2.4

libapache2-mod-auth-openidc
jakub-onderka/openid-connect-php:1.0.0-rc1

**Comment by bita-kea** [comment]
Same here.

Our MISP has a lot of organisations. And each login (via keycloak) of existing users changes their organisation to the one specified in ‘default_org’.
Is there any way to specify that the existing organisation in MISP would be prioritised?
Or specify the organisation as we do with roles?
So, then add the organisation field in the keycloak profile?

For context:
Keycloak v26

MISP v2.5
Ubuntu 24
php v8.3
apache2 v2.4

libapache2-mod-auth-openidc
jakub-onderka/openid-connect-php:1.0.0-rc1

**Comment by bk1b** [comment]
Same here.

Our MISP has a lot of organisations. And each login (via keycloak) of existing users changes their organisation to the one specified in ‘default_org’.
Is there any way to specify that the existing organisation in MISP would be prioritised?
Or specify the organisation as we do with roles?
So, then add the organisation field in the keycloak profile?

For context:
Keycloak v26

MISP v2.5
Ubuntu 24
php v8.3
apache2 v2.4

libapache2-mod-auth-openidc
jakub-onderka/openid-connect-php:1.0.0-rc1

**Comment by iglocska** [comment]
From what I can tell: it sets (and enforces) the default as long as the correct values are not passed in the oidc jwt. The fields that should be included are:

organization
organization_uuid

If these are provided by keycloak then it will be enforced in MISP, rather than the hard coded default.

**Comment by bk1b** [comment]
Yes! I write here my solution in case someone finds the same problem:

1/ At keycloak:
Create a new Attribute at keycloak by going to "Realms settings" > "User profile" > "Create attribute" > give the name: organization and fill in other fields as needed.

Then, using the same organizations names from MISP, set the correct organization to each user.

Go to "Clients" > misp.client > "Client scopes" > misp.client-dedicated > "Add mapper" > "By configuration" > "User attribute" > chose "organization" and fill in other fields as needed.

It is not need it to include UUID attribute.
Organization feature was not working for me.

2/ At MISP:
No changes. It did not work for me when I tried to include organization at the "OIDCScope" or "scopes".

**Comment by righel** [comment]
Correct solution posted by @bk1b and no feedback received.
